protocol/superchainerc20/superc20Redesign.md

```diff
@@ -0,0 +1,101 @@
+    - The `OptimismSuperchainERC20Factory` will of course be modified to deploy the new simpler implementations, but will still record deployments in a mapping.
+
+The updated flow would look like follows:
+
+```mermaid
+sequenceDiagram
+  participant from
+  participant L2SBA as L2StandardBridge (Chain A)
+  participant SuperERC20_A as SuperchainERC20 (Chain A)
+  participant Messenger_A as L2ToL2CrossDomainMessenger (Chain A)
+  participant Inbox as CrossL2Inbox
+  participant Messenger_B as L2ToL2CrossDomainMessenger (Chain B)
+  participant L2SBB as L2StandardBridge (Chain B)
+  participant SuperERC20_B as SuperchainERC20 (Chain B)
+  participant Recipient as to
+
+  from->>L2SBA: sendERC20To(tokenAddr, to, amount, chainID)
+  L2SBA->>SuperERC20_A: burn(from, amount)
+  L2SBA->>Messenger_A: sendMessage(chainId, message)
+  L2SBA-->L2SBA: emit SentERC20(tokenAddr, from, to, amount, destination)
+  Inbox->>Messenger_B: relayMessage()
+  Messenger_B->>L2SBB: relayERC20(tokenAddr, from, to, amount)
+  L2SBB->>SuperERC20_B: mint(to, amount)
+  L2SBB-->L2SBB: emit RelayedERC20(from, to, amount, source)
+
+```
+
+It’s important to notice that, even though this functions get implemented in the bridge, tokens will still have agency to reimplement the bridging functions on their contract if they want so. The alternative design gives the optionality.
+
+## Comparison
+
+### Pros of using the alternative design
+
+- **Compatible with the current design:** adding the `sendERC20` and `relayERC20` functions to the bridge does not imply that these should no longer exist in the token. It only gives the option to token developers not to include them.
+- **Separation of functionalities:** bridging lives in the bridge, token functions live in the token. This fosters clarity but also comes with improved upgrade flows.
+    - If bridging functions are implemented on the ERC20, updates to the bridge might require updates to every ERC20. This might be easy to do for implementations that use a BeaconProxy structure, but not in other cases.
+- **Easier for integrators:** integrators must deal with a single entrypoint instead of many. This would improve their devX considerably.
+- **Improved indexing and monitoring:** only monitoring events and traces coming from a single source would be necessary instead of many. Improved monitoring correlates with improved security.
+- **Improved backwards compatibility:** the only modifications required to implement interop for a token are including `remoteToken` and giving mint and burn permissions to the `L2StandardBridge`, which can be implemented before the hardfork. This would not be possible with the current design. 
+- **Homogeneous implementation:** most tokens on other chains do not implement bridging functionalities (except for OFTs and some others), which leads to a fragmented developer experience.
+
+### Cons of using the alternative design
+
+- **Timing:**
```

Comment:
Any cons of the design itself, or do we feel this is strictly superior? I could be wrong but I thought we did consider this approach originally (since xERC20 also has its methods on the bridge contract), so I'm curious why we didn't like it originally and what changed

Reply:
We did originally think of this approach but it was discarded in favor of putting the interface on the token, this decision was made towards the early stages of the project

Reply:
We thought about this and could not identify any strong con tbh. I do recognize there is some beauty on having the function directly accesible in the token, like a native behaviour of the token itself. But couldn't formalize that into a proper argument. 
Happy to open up the discussion though.

Reply:
The main downside I can think of is that it may be a less familiar or less intuitive design—normally to interact with a token, you call the token contract, but now you call a different contract. xERC20 is similar though, so that precedent helps bring the UX back towards something familiar. So overall I am not worried about that downside.

I am curious why we discarded this idea originally though. I think it might have just been for that UX reason?

Reply:
> I am curious why we discarded this idea originally though. I think it might have just been for that UX reason?

It was originally discarded because we had strong intuitions on defining it as an interface originally and after learning more we dropped those desires